internal/controller/postgrescluster/delete_test.go

```diff
@@ -412,27 +402,20 @@ func TestReconcilerHandleDeleteNamespace(t *testing.T) {
 				client.Merge.Type(), []byte(`{"metadata":{"finalizers":[]}}`)))))
 	})
 
-	var instances []appsv1.StatefulSet
-	var ready int32
-	assert.NilError(t, wait.Poll(time.Second, Scale(time.Minute), func() (bool, error) {
-		list := appsv1.StatefulSetList{}
-		selector, err := labels.Parse(strings.Join([]string{
-			"postgres-operator.crunchydata.com/cluster=" + cluster.Name,
-			"postgres-operator.crunchydata.com/instance",
-		}, ","))
-		assert.NilError(t, err)
-		assert.NilError(t, cc.List(ctx, &list,
-			client.InNamespace(cluster.Namespace),
-			client.MatchingLabelsSelector{Selector: selector}))
-
-		instances = list.Items
-
-		ready = 0
-		for i := range instances {
-			ready += instances[i].Status.ReadyReplicas
-		}
-		return ready >= 2, nil
-	}), "expected 2 instances to be ready, got:\n%v", ready)
+	// Wait until instances are healthy.
+	if ready := int32(0); !assert.Check(t,
+		wait.Poll(time.Second, Scale(time.Minute), func() (bool, error) {
```

Comment:
These `if !Check(…) { FailNow() }` look awkward. I put some logging ahead of the `FailNow` while debugging flakes, but that code was grotty and not ready to commit.